app/javascript/controllers/unsaved_controller.js

```diff
@@ -98,4 +101,22 @@ export default class extends Controller {
       }
     }
   }
+
+  hasChangedFields() {
+    // scan for changes in form (ignoring slim select, which takes a bit to update)
+    const changed = this.element.querySelectorAll(".is-changed:not(.ss-main):not([style='display:none'])")
```

Comment:
This will select all elements with the `is-changed` class that does not also have the `.ss-main` class and is not hidden with `display:none`
Then, we loop through all of those, and if any are not `slimselect` or `offsetParent`, then it is true that there changed fields.

Reply:
Ooh, I left several unnecessary things in this call.  Should just be `this.element.querySelectorAll(".is-changed")`.  Will fix.

But you're right - i loop through to check them all.